torch/csrc/jit/script/compiler.cpp

```diff
@@ -29,6 +29,113 @@ using ValueTable = std::unordered_map<std::string, SugaredValuePtr>;
 using AttributeMap = std::unordered_map<std::string, Const>;
 using ListAttributeMap = std::unordered_map<std::string, std::vector<Const>>;
 
+struct Refinements {
+
+  //using ordered map for deterministic graph output
+  std::map<std::string, TypePtr> types;
```

Comment:
Can you use a small struct here? There are about 10 lines below where types and ranges get repeated. It will get worse if someone needs to add something later. It also stores two copies of every variable name, two entire red-black trees, and doubles the number of traversals!